### PR TITLE
mysql root pwd for rh and suse

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: states/_dbmgmt.sls [hunk 1]
```diff
@@ -8,6 +8,15 @@
 include:
   - mysql._salt
 
+{% if salt['grains.get']('os_family') in ['RedHat', 'Suse' ] %}
+{{ comp_type }}_rootpwd:
+  cmd.run:
```
Comment (arnisoph) [inline states/_dbmgmt.sls:13]:
```
cmd:
  - run
```

please :)

File: states/_dbmgmt.sls [hunk 2]
```diff
@@ -8,6 +8,15 @@
 include:
   - mysql._salt
 
+{% if salt['grains.get']('os_family') in ['RedHat', 'Suse' ] %}
```
Comment (arnisoph) [inline states/_dbmgmt.sls:11]: Is this RedHat/ Suse specific? AFAIR when you install mysql/ mariadb on Debian/ RedHat root doesn't have a password. Then you can use https://github.com/bechtoldt/saltstack-mysql-formula/blob/master/states/_dbmgmt.sls#L38 to set a new one. But I might be wrong on this.

**Comment by hoonetorg** [comment]
AFAIU 
on debian-based you set the initial mysql-root-pw with debconf, if i got it right. that means on debian you have a mysql-root-pwd directly after package installation, don't you? 
this is not possible on rpm-based. so after rpm install, you have no mysql-root-pw. you need to initially set it.
if you use connection details with password (not over socket) for https://github.com/bechtoldt/saltstack-mysql-formula/blob/master/states/_dbmgmt.sls#L38, it will fail, because: i the connection details include the not-yet-set mysql-root-pwd. in german we say "henne-ei-problem" (don't know the correct english phrase for that).

**Comment by arnisoph** [comment]
@hoonetorg Right, now I see the problem. I've used this formula on Debian only so far, so I didn't realized the problem on RHEL.

I'm going to accept this pull request (thanks for that!) but will add some style/convience-related changes.

**Pull request merged**